Source/levels/drlg_l1.cpp

```diff
@@ -940,38 +940,28 @@ void Substitution()
 {
 	for (int y = 0; y < DMAXY; y++) {
 		for (int x = 0; x < DMAXX; x++) {
-			if (FlipCoin(4)) {
-				uint8_t c = TileDecorations[dungeon[x][y]];
-				if (c != 0 && !Protected.test(x, y)) {
-					int rv = GenerateRnd(16);
-					int i = -1;
-					while (rv >= 0) {
-						i++;
-						if (i == sizeof(TileDecorations)) {
-							i = 0;
-						}
-						if (c == TileDecorations[i]) {
-							rv--;
-						}
-					}
-
-					// BUGFIX: Add `&& y > 0` to the if statement. (fixed)
-					if (i == VWall4 && y > 0) {
-						if (TileDecorations[dungeon[x][y - 1]] != VWall2 || Protected.test(x, y - 1))
-							i = VWall2;
-						else
-							dungeon[x][y - 1] = VWall5;
-					}
-					// BUGFIX: Add `&& x + 1 < DMAXX` to the if statement. (fixed)
-					if (i == HWall4 && x + 1 < DMAXX) {
-						if (TileDecorations[dungeon[x + 1][y]] != HWall2 || Protected.test(x + 1, y))

```

Comment:
Is this really dead code?
With seed 1213599971 (cathedral) this is executed?
🤔

Reply:
How sure are you about that?

VWall4
![image](https://user-images.githubusercontent.com/204594/182045309-7b3d92be-2063-49bf-ac3a-fcfa1f3c4f86.png)

HWall4
![image](https://user-images.githubusercontent.com/204594/182045316-484ae973-6497-4868-8822-fc9170cfac06.png)

Maybe you where generating level 2 and using my test fixtures?
![image](https://user-images.githubusercontent.com/204594/182046039-b8451491-76fb-4388-8ed4-63def4a01cd1.png)

Reply:
1213599971 (level 1 no quests):
![image](https://user-images.githubusercontent.com/204594/182046080-bbb8c1d5-119f-47a0-bf75-8f392b9cfd9a.png)

1213599971 (level 2 butcher, no poison water)
![image](https://user-images.githubusercontent.com/204594/182046144-e75052a5-f639-456d-986d-61f79c14f362.png)

Reply:
I triggered it in dungeon level 2 with poison water.

Master:
![grafik](https://user-images.githubusercontent.com/25415264/182077853-4cc20e72-3d2f-4fa7-bb5d-d4840853627a.png)

PR:
![grafik](https://user-images.githubusercontent.com/25415264/182078146-efadfbbf-b413-435f-8ec5-cbb1c051e96f.png)

Save game:
[single_33.zip](https://github.com/diasurgical/devilutionX/files/9231115/single_33.zip)

Reply:
Interesting that it's not marked as protected somehow :)